src/components/Api.js

```diff
@@ -0,0 +1,118 @@
+export default class Api {
+  // Constructor initializes API with a base URL and authorization headers
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  // Method to check the server's response and handle errors
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json(); // Return JSON if response is OK
+    }
+    return Promise.reject(`Error: ${res.status}`); // Reject promise with error status
+  }
+
+  // General error handler for catching and logging errors
+  _handleError(err) {
+    console.error(err); // Log error to console
+    return Promise.reject(err); // Reject the promise with the error
+  }
+
+  // Fetch initial set of cards from the server
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers, // Pass authorization headers
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json(); // Return JSON if response is OK
+        }
+        return Promise.reject(`Error: ${res.status}`); // Reject if there's an error
+      })
+      .catch((err) => {
+        console.error("API error:", err); // Log the error
+        return Promise.reject(err); // Propagate the error
+      });
+  }
+
+  // Fetch user information from the server
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers, // Authorization headers for the request
+    })
+      .then(this._checkResponse) // Handle response validation
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.